src/Core/Domain/Cart/QueryHandler/GetProductQuantityInCartHandler.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Cart\QueryHandler;
+
+use Doctrine\DBAL\Connection;
+use PDO;
+use PrestaShop\PrestaShop\Core\Domain\Cart\Exception\CartException;
+use PrestaShop\PrestaShop\Core\Domain\Cart\Query\GetProductQuantityInCart;
+
+/**
+ * Handles GetProductQuantityInCart query using dbal connection
+ */
+final class GetProductQuantityInCartHandler implements GetProductQuantityInCartHandlerInterface
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+    /**
+     * @var string
+     */
+    private $dbPrefix;
+
+    /**
+     * @param Connection $connection
+     * @param string $dbPrefix
+     */
+    public function __construct(
+        Connection $connection,
+        string $dbPrefix
+    ) {
+        $this->connection = $connection;
+        $this->dbPrefix = $dbPrefix;
+    }
+
+    /**
+     * @param GetProductQuantityInCart $query
+     *
+     * @return int
+     */
+    public function handle(GetProductQuantityInCart $query): int
```

Comment:
Besides it appears this handler is now useless (see CartController)

Reply:
Yes and no 😅

- indeed this Query was built before we had the debate about Queries VS data providers and now that we have discussed that it would make a lot more sense to have a repository search function
- although it's being used directly into a controller action 🤔so it could make sense if this is a standard query required often (maybe we could improve the Query to make it more reusable like GetCartAmounts)

However as you stated ... this Query is now useless 😅so debate is close, I delete it.